opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -307,12 +307,16 @@ namespace Opm
                     
                     const bool hasGroupControl = this->isInjector() ? inj_controls.hasControl(Well::InjectorCMode::GRUP) :
                                                                       prod_controls.hasControl(Well::ProducerCMode::GRUP);
-                    bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP; 
-                    if (! (isGroupControl && !this->param_.check_group_constraints_inner_well_iterations_)) {
-                        changed = this->checkIndividualConstraints(ws, summary_state, deferred_logger, inj_controls, prod_controls);
-                    }
-                    if (hasGroupControl && this->param_.check_group_constraints_inner_well_iterations_) {
-                        changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
+                    const bool isGroupControl = ws.production_cmode == Well::ProducerCMode::GRUP || ws.injection_cmode == Well::InjectorCMode::GRUP;
+
+                    if (this->param_.check_group_constraints_inner_well_iterations_) {
+                        if (!isGroupControl && hasGroupControl) {
+                            changed = changed || this->checkGroupConstraints(well_state, group_state, schedule, summary_state,deferred_logger);
```

Comment:
Not a big deal, but since changed here is always false this could be simplified (remove 'changed ||' on the rhs)